### PR TITLE
SSO: Show modal on disconnection

### PR DESCRIPTION
**Pull request by heavyweight**
This PR adds `SurveyModal` component and uses that to display a modal when users disable sso

Related to #

## Proposed Changes

* Add SurveyModal component
* Wire it to be used in sso

## Testing Instructions
- Build the PR or use the live link
- Go to /settings/security/{someAtomicSite}
- Disable sso
- You should see a modal that leads to the sso survey

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/survey-modal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { Gridicon } from '@automattic/components';
+import { Button } from '@wordpress/components';
+import classNames from 'classnames';
+import cookie from 'cookie';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
```
Comment (escapemanuele) [inline client/components/survey-modal/index.jsx]: Can we use the one from @wordpress?

**Comment by escapemanuele** [comment]
This is great, thanks!

Let's show the modal only for en users for now (as the CrowdSignal survey will be in English)! And create the survey on Crowdsignal before shipping this :D

**Comment by javierarce** [comment]
It looks very nice, @heavyweight, just two things.

The "Remind later" should also appear in blue. 

Screenshot 2024-03-15 at 15 14 37

A few hours ago, we made some changes to the design of those pop-ups so that when they display an image, we don't include the top header. Sorry for this last-minute change!

image

Link to the design: eqlepL7MHifK8Bgr1OWqLr-fi-15013_60324

**Comment by heavyweight** [comment]
> It looks very nice, @heavyweight, just two things.
> 
> The "Remind later" should also appear in blue.
> 
> Screenshot 2024-03-15 at 15 14 37
> A few hours ago, we made some changes to the design of those pop-ups so that when they display an image, we don't include the top header. Sorry for this last-minute change!
> 
> image
> 
> Link to the design: [eqlepL7MHifK8Bgr1OWqLr-fi-15013_60324](https://href.li/?https://www.figma.com/file/eqlepL7MHifK8Bgr1OWqLr/?node-id=15013_60324)

Hi Javier,
Fixed the cancel link now. Had a typo.

File: client/components/survey-modal/style.scss [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			height: 0;
+
+			img {
+				width: 100%;
+				display: block;
+			}
+		}
+
+		.modal-survey-notice__popup-content {
+			padding: 18px 24px 30px;
```
Comment (escapemanuele) [inline client/components/survey-modal/style.scss]: This padding should be 16px

**Review by escapemanuele — APPROVED** [review]
LGTM!

File: client/components/survey-modal/style.scss (hunk 2 above)
Comment (agrullon95) [reply client/components/survey-modal/style.scss]: Fixed

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11607014

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @heavyweight, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `SSO Survey`
- `Spare a moment? We'd love to hear why you want to disable SSO in a quick survey.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11607014).

**Pull request merged**